partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/RelConverter.kt

```diff
@@ -607,13 +617,32 @@ internal object RelConverter {
         // only rewrite top-level SFW
         override fun visitExprSFW(node: Expr.SFW, ctx: Context): AstNode = node
 
-        override fun visitExprAgg(node: Expr.Agg, ctx: Context) = ast {
-            val id = identifierSymbol {
-                symbol = syntheticAgg(ctx.aggregations.size)
-                caseSensitivity = org.partiql.ast.Identifier.CaseSensitivity.INSENSITIVE
+        override fun visitExprCall(node: Expr.Call, ctx: Context) = ast {
+            // TODO replace w/ proper function resolution to determine whether a function call is a scalar or aggregate.
+            //  may require further modification of SPI interfaces to support
+            when (node.function.isAggregateCall()) {
+                true -> {
+                    val id = identifierSymbol {
+                        symbol = syntheticAgg(ctx.aggregations.size)
+                        caseSensitivity = org.partiql.ast.Identifier.CaseSensitivity.INSENSITIVE
+                    }
+                    ctx.aggregations += node
+                    exprVar(id, Expr.Var.Scope.DEFAULT)
+                }
+                else -> node
+            }
+        }
+
+        private fun String.isAggregateCall(): Boolean {
```

Comment:
(self-review) since the function resolution specification for scalar vs aggregate functions has not yet been defined, opt to hard-code the aggregation logic to the planner for now. Once we formalize the function resolution, we can remove this hard-coded logic.